### PR TITLE
[DLP] Update an existing job trigger

### PR DESCRIPTION
**Pull request by bhavika-dhanwani**
Implementation of [Updating an existing job trigger](https://cloud.google.com/dlp/docs/creating-job-triggers#update_an_existing_job_trigger)

## Checklist

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+              .build();
+
+      // Combine configurations into a request for the service.
+      UpdateJobTriggerRequest updateJobTriggerRequest = UpdateJobTriggerRequest.newBuilder()
+              .setName(JobTriggerName.of(projectId, jobTriggerName).toString())
+              .setJobTrigger(jobTrigger)
+              .setUpdateMask(fieldMask)
+              .build();
+
+      // Send the scan request and process the response
```
Comment (soumya92) [inline dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java:80]: Update this comment; we are not sending a scan request, just updating a job trigger.

Job Trigger updates do not immediately cause scans, and job trigger results are not returned in api but need to be fetched separately later.

File: dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      JobTrigger jobTrigger = JobTrigger.newBuilder()
+              .setInspectJob(inspectJobConfig)
+              .build();
+
+      FieldMask fieldMask = FieldMask.newBuilder()
+              .addPaths("inspect_job.inspect_config.info_types")
+              .addPaths("inspect_job.inspect_config.min_likelihood")
+              .build();
+
+      // Combine configurations into a request for the service.
```
Comment (soumya92) [inline dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java:73]: "// Update the job trigger with the new configuration"

File: dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      // Specify the type of info the inspection will look for.
+      // See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types
+      InfoType infoType = InfoType.newBuilder().setName("PERSON_NAME").build();
+
+      InspectConfig inspectConfig = InspectConfig.newBuilder()
+              .addInfoTypes(infoType)
+              .setMinLikelihood(Likelihood.LIKELY)
+              .build();
+
+      // Configure the inspection job we want the service to perform.
```
Comment (soumya92) [inline dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java:59]: We should just remove this comment, I don't think it's adding any value.

**Review by soumya92 — CHANGES_REQUESTED** [review]

File: dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      InspectJobConfig inspectJobConfig = InspectJobConfig.newBuilder()
+              .setInspectConfig(inspectConfig)
+              .build();
+
+      JobTrigger jobTrigger = JobTrigger.newBuilder()
+              .setInspectJob(inspectJobConfig)
+              .build();
+
+      FieldMask fieldMask = FieldMask.newBuilder()
```
[the remark this answers is not in the document]
Comment (Sita04) [reply dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java:68]: +1 A doc link on constructing these paths would also be helpful

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (bhavika-dhanwani) [reply dlp/snippets/src/main/java/dlp/snippets/TriggersPatch.java:70]: > +1 A doc link on constructing these paths would also be helpful

Done

**Review by soumya92 — APPROVED** [review]

**Review by Sita04 — APPROVED** [review]

**Pull request merged**